core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -109,20 +109,17 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
 
       new Thread(Utils.getFormattedClassName(this) + "-mesos-driver") {
         setDaemon(true)
+        setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler {
```

Comment:
I only see it one place, in `FsHistoryProvider`, where it looks like it's mostly there to support testing.